### PR TITLE
[flang] Add a GitHub action for SNAP

### PR DESCRIPTION
**Pull request by banach-space**

### CODE REVIEW

**Review by AlexisPerry — APPROVED** [review]
This looks good to me, thanks for adding it!  CI is a wonderful thing, in my opinion.  Please wait for approval from at least one other person before merging.

File: .github/workflows/flang-tests-snap.yml
Comment (schweitzpgi) [inline .github/workflows/flang-tests-snap.yml:47]: It would be nice to not have to build classic flang for pgmath. Someday.

**Review by schweitzpgi — APPROVED** [review]

**Comment by sscalpone** [comment]
How do we decide what belongs in pre-commit ci & what belongs in test-suites?

**Comment by clementval** [comment]
@banach-space Is that 2hours on the top of the OS X build or will it run in parallel. 
@sscalpone I think it belongs in test-suites once they can run it upstream.

**Comment by banach-space** [comment]
@clementval These workflows are independent and each defines only 1 job. GitHub [allows](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) us 20 concurrent jobs. So with 2 jobs, we can be confident that our workflows will run in parallel.

@sscalpone I agree with Valentin here, but [llvm-test-suite](https://github.com/llvm/llvm-test-suite) is not possible ATM. There's definitely no intention to keep this config "on" indefinitely. In fact, I intend to add SNAP to llvm-test-suite soon. However, we won't be able to use it until:
* upstreaming is complete
* there's CMake support for LLVM Flang
* https://github.com/lanl/SNAP/pull/19 is merged

So, this is our best option at the moment.

**Comment by schweitzpgi** [comment]
Enabling for now. Can always back this out if it is a headache.

**Pull request merged**
